code/modules/species/station/utility_frame.dm

```diff
@@ -116,4 +121,98 @@
 	name = "Frame Head Plating"
 	body_parts = list(BP_HEAD)
 
+/mob/living/carbon/human/proc/detach_limb()
+	set category = "IC"
+	set name = "Detach Limb"
+	set desc = "Detach one of your robotic appendages."
+
+	if(last_special > world.time)
+		return
+
+	if(stat || paralysis || stunned || weakened || lying || restrained())
+		to_chat(src, SPAN_WARNING("You can't do that in your current state!"))
+		return
+
+	var/obj/item/organ/external/E = get_organ(zone_sel.selecting)
+
+	if(!E)
```

Comment:
`if(!E || E.is_stump())` - or handle stumps in their own block ('your X is too damaged to remove')